vilebot/src/test/java/vilebot/handlers/user/TransferKarmaTest.java

```diff
@@ -0,0 +1,138 @@
+import org.pircbotx.Channel;
+import org.pircbotx.User;
+import org.pircbotx.hooks.events.MessageEvent;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class TransferKarmaTest
+{
+    private MessageEvent event;
+
+    private static final String VILEBOT_CONFIG_FILE = "cfg/vilebot.conf";
+
+    private static final String VILEBOT_CONFIG_FILE_SECONDARY = "cfg/vilebot.conf.example";
+
+    @Before
+    public void setup()
+    {
+        event = mock( MessageEvent.class );
+        User user = mock( User.class );
+        Channel channel = mock( Channel.class );
+        when( event.getUser() ).thenReturn( user );
+        when( user.getNick() ).thenReturn( "senderNick" );
+        when( event.getChannel() ).thenReturn( channel );
+        when( channel.getName() ).thenReturn( getChannel() );
+    }
+
+    @Test
+    public void invalidChannelTest()
+    {
+        Channel channel = mock( Channel.class );
+        when( event.getChannel() ).thenReturn( channel );
+        when( channel.getName() ).thenReturn( "INVALID CHANNEL" );
+
+        String ircmsg = "!transfer receiver 10";
+        when( event.getMessage() ).thenReturn( ircmsg );
+
+        KarmaTransfer karmaTransfer = new KarmaTransfer();
+        karmaTransfer.onGenericMessage( event );
+
+        String expectedResponse = "You must be in " + getChannel() + " to make or accept wagers.";
+        verify( event, times( 1 ) ).respondWith( expectedResponse );
+    }
+
+    @Test
+    public void transferToSelfTest()
+    {
+        String ircmsg = "!transfer senderNick 10";
+        when( event.getMessage() ).thenReturn( ircmsg );
+
+        KarmaTransfer karmaTransfer = new KarmaTransfer();
+        karmaTransfer.onGenericMessage( event );
+
+        String expectedResponse = "Really? What's the point of transferring karma to yourself?";
+        verify( event, times( 1 ) ).respondWith( expectedResponse );
+    }
+
+    @Test
+    public void invalidTransferCancelTest()
+    {
+        String ircmsg = "!transfercancel";
+        when( event.getMessage() ).thenReturn( ircmsg );
+
+        KarmaTransfer karmaTransfer = new KarmaTransfer();
+        karmaTransfer.onGenericMessage( event );
+
+        verify( event, times( 1 ) ).respondWith( "No active transfer."
+            + " To transfer karma enter '!transfer <noun> <karma amount>'." );
+    }
+
+    @Test
+    public void invalidRejectTransferTest()
+    {
+        String ircmsg = "!reject";
+        when( event.getMessage() ).thenReturn( ircmsg );
+
+        KarmaTransfer karmaTransfer = new KarmaTransfer();
+        karmaTransfer.onGenericMessage( event );
+
+        verify( event, times( 1 ) ).respondWith( "No active transfer."
+            + " To transfer karma enter '!transfer <noun> <karma amount>'." );
+    }
+
```

Comment:
No test for valid transfer?

Reply:
Testing the happy path involves creating an integration test that sets up database interactions, which is not a trivial task. There have been previous attempts afaik (see [this issue](https://github.com/oldterns/VileBot/issues/52)). Considering the current state of the project, setting up this test is outside the scope of this PR. 

The VileBot1 way is to test in production. /s

Reply:
In `PodIRCBot`, the bot get messages from stdin, and send messages to stdout. Additionally, it easy to configure a new location for the redis server due to the nature of containers. One can probably start a test redis server using this (https://pypi.org/project/testing.redis/), and get the state of Redis by querying the database.

Reply:
Sounds promising. Looking forward to the refactor. Resolving this issue for now.